fastn-core/src/commands/serve.rs

```diff
@@ -191,6 +191,15 @@ pub async fn serve_helper(
     req: fastn_core::http::Request,
     only_js: bool,
 ) -> fastn_core::Result<fastn_core::http::Response> {
+    #[cfg(feature = "auth")]
+    if let Ok(auth_enabled) = std::env::var("FASTN_ENABLE_AUTH") {
+        // only apply migrations when FASTN_ENABLE_AUTH=true
+        if auth_enabled == "true" {
+            tracing::info!("running auth related migrations");
+            fastn_core::commands::serve::enable_auth()?
+        }
+    }
```

Comment:
Problem is this gets called on every HTTP request, and we do not want to run migration on every HTTP request. We have to explicitly call this method from `hostn` when `fastn_ds::DocumentStore()` is getting constructed for any domain.